datastore/datastore.go

```diff
@@ -679,3 +701,16 @@ func (c *Client) Mutate(ctx context.Context, muts ...*Mutation) (ret []*Key, err
 	}
 	return ret, nil
 }
+
+// ReadOptions provides specific instructions for how to access documents in the database.
+// Currently, only ReadTime is supported.
+type ReadOptions struct {
+	ReadTime time.Time
+}
+
+// WithReadOptions specifies constraints for accessing documents from the database,
+// e.g. at what time snapshot to read the documents.
+// Once set, all subsequent reads from the client use these options.
```

Comment:
What if a client wants to a read, and then in the future do another read? This comment makes me think this method cannot be called twice when in reality, it probably could be adjusted as needed. Perhaps rephrase as, "The client uses this value for subsequent reads"

Reply:
Done.